message/getToken.go

```diff
@@ -11,6 +12,10 @@ func (t Token) String() string {
 	return hex.EncodeToString(t)
 }
 
+func (t Token) Hash() uint64 {
```

Comment:
We could cache the value if the function is used often, eg:

```
type Token struct {
  data []byte
  hash uint64
}
```

Calculate the value on first use, save it and then use only cached value.

Reply:
Hash is a very cheap operation so was can leave it as is otherwise we need to introduce a `constructor` which fills the value.